Documentation/Troubleshooting.md

```diff
@@ -58,3 +58,7 @@ extensions, or you try to use VirtualBox without using a x64 virtualization mode
 - If booting on bare metal, your CPU is too old to boot Serenity.
 - If you're using VirtualBox, you need to enable PAE/NX. Check the instructions [here.](VirtualBox.md)
 - If you're using QEMU, the [CPU model configuration](https://qemu-project.gitlab.io/qemu/system/qemu-cpu-models.html) is not exposing PAE.
+
+### Boot fails with "KVM doesn't support guest debugging"
+- Try to update your host kernel.
```

Comment:
The important thing is that on the system that built QEMU, the constant `KVM_CAP_SET_GUEST_DEBUG` is defined in ``<linux/kvm.h>``, and on the system that's running QEMU, ``ioctl(fd, KVM_CHECK_EXTENSION, KVM_CAP_SET_GUEST_DEBUG)`` on the KVM device returns 1.

It looks like there were some bugs in the kernel around 2020 that were fixed in this patch: https://lore.kernel.org/all/20200507115011.494562-2-pbonzini@redhat.com/

As long as you're not using an ancient/stable distro like Debian stable or Void, simply rebuilding QEMU from source using our Toolchain/BuildQemu.sh script will probably re-enable the feature.

Reply:
QEMU checks both of those here: <https://gitlab.com/qemu-project/qemu/-/blob/222059a0fccf4af3be776fe35a5ea2d6a68f9a0b/accel/kvm/kvm-all.c#L2540>

The kernel IOCTL is here, but I'll leave the archaeology of checking which kernel versions it appears in to the reader :) <https://elixir.bootlin.com/linux/v6.1.1/source/arch/x86/kvm/x86.c#L4396>

Reply:
Thanks for your feedback.

In my case, the build and run host is the same (source-based distro). I checked the linux kernel history and the [commit](https://github.com/torvalds/linux/commit/b9b2782cd5) which fixed handling `KVM_CAP_SET_GUEST_DEBUG` was released first in kernel 5.10. This checks out for my system.